Automattic/pocket-casts-android
# Playback 2023: Skip "Plus stories" in sequence

| 📘 Part of: #1463 |
|:---:|

This PR adds a mechanism to:

1. Set which stories are Plus or not;
2. Skip a bunch of Plus stories in sequence for a non-paid user
3. Skip a bunch of Plus stories in sequence for a non-paid user when going to previous stories

Basically — given the "Plus stories" will have an upsell, we don't want users to see the same story multiple times. So we just display it once, whether is going to next stories or previous one.

## To test

Before testing, add `override val plusOnly: Boolean = true` in `StoryYearOverYear` and `StoryCompletionRate`.
Note that the first plus story "StoryYearOverYear" doesn't have an upsell yet.

Then:

1. Run the app
2. Login with `Free` account
3. Let the stories run (don't tap to skip to the next or to the previous one)
4. ✅ After the "YearOverYear" story is shown, it should skip to the last one
5. Tap to go to the previous story
6. ✅ It should go back to the "YearOverYear" (skipping CompletionRate story)
7. On the "YearOverYear" story, tap to go next
8. ✅ It should go to the last story
9. Exit the stories
10. Assign `Plus` plan to account
11. Tap "Refresh now" on Profile tab
12. Repeat the testing steps
13. ✅ However, all stories should show (no stories will be skipped)
14. Assign `Patron` plan to account
15. Tap "Refresh now" on Profile tab
16. Repeat the testing steps
17. ✅ All stories should show (no stories will be skipped)

** Tests cover skip scenarios when there are multiple plus stories in between

## Checklist
 
#### I have tested any UI changes...
